src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -760,6 +819,18 @@ public override Task OpenAsync(CancellationToken cancellationToken)
                 TaskCompletionSource<DbConnectionInternal> completion = new TaskCompletionSource<DbConnectionInternal>();
                 TaskCompletionSource<object> result = new TaskCompletionSource<object>();
 
+                result.Task.ContinueWith((t) =>
+                {
+                    if (t.Exception != null)
+                    {
+                        s_diagnosticListener.WriteConnectionOpenError(operationId, this, t.Exception);
+                    }
+                    else
+                    { 
+                        s_diagnosticListener.WriteConnectionOpenAfter(operationId, this);
+                    }
+                });
```

Comment:
You should pass TaskScheduler.Default as the scheduler argument here. This continuation should also only be created if diagnostic logging is enabled.